06_HelloGraphicsQueue/main.cpp

```diff
@@ -150,10 +139,21 @@ class HelloGraphicsQueueApp final : public examples::MonoDeviceApplication, publ
 		imgLayoutTransitionBarrier.newLayout = asset::IImage::EL_TRANSFER_DST_OPTIMAL;
 		imgLayoutTransitionBarrier.barrier.dstAccessMask = E_ACCESS_FLAGS::EAF_MEMORY_WRITE_BIT;
 
-		IGPUCommandBuffer::SImageMemoryBarrier imgClearBarrier = constructDefaultInitializedImageBarrier();
-		imgClearBarrier.oldLayout = asset::IImage::EL_TRANSFER_DST_OPTIMAL;
-		imgClearBarrier.barrier.srcAccessMask = E_ACCESS_FLAGS::EAF_MEMORY_WRITE_BIT;
+		auto generateNextBarrier = [](
+			const IGPUCommandBuffer::SImageMemoryBarrier& prevoiusBarrier,
+			IImage::E_LAYOUT newLayout,
+			asset::E_ACCESS_FLAGS dstAccessMast) -> IGPUCommandBuffer::SImageMemoryBarrier
+			{
+				IGPUCommandBuffer::SImageMemoryBarrier ret = prevoiusBarrier;
+				ret.oldLayout = prevoiusBarrier.newLayout;
+				ret.newLayout = newLayout;
+				ret.barrier.srcAccessMask = prevoiusBarrier.barrier.dstAccessMask;
+				ret.barrier.dstAccessMask = dstAccessMast;
+
+				return ret;
+			};
 
+		// Using `float32` union member since format of used images is E_FORMAT::EF_R8G8B8A8_SRGB.
```

Comment:
add "SRGB, which is a fixed-point format and those are encoded/decoded from/to float"